nemo/backends/pytorch/actions.py

```diff
@@ -1334,7 +1344,11 @@ def train(
                         final_loss.backward(bps_scale.to(final_loss.get_device()))
                     # single device (CPU or GPU)
                     else:
-                        final_loss.backward(bps_scale.to(final_loss.get_device()))
+                        # Fix (workaround?) enabling to backpropagate gradiens on CPUs.
+                        if final_loss.get_device() < 0:
+                            final_loss.backward(bps_scale.to(final_loss))
```

Comment:
this is a bug: you should not be doing tenosr.to(tensor) - please fix

Reply:
There is a separate PR:
https://github.com/NVIDIA/NeMo/pull/308
Let's move discussion there.

Reply:
fixed - here and there